js/ftx.js

```diff
@@ -911,26 +911,23 @@ module.exports = class ftx extends Exchange {
         const timestamp = this.parse8601 (this.safeString (trade, 'time'));
         const priceString = this.safeString (trade, 'price');
         const amountString = this.safeString (trade, 'size');
-        const price = this.parseNumber (priceString);
-        const amount = this.parseNumber (amountString);
-        const cost = this.parseNumber (Precise.stringMul (priceString, amountString));
         if ((symbol === undefined) && (market !== undefined)) {
             symbol = market['symbol'];
         }
         const side = this.safeString (trade, 'side');
         let fee = undefined;
-        const feeCost = this.safeNumber (trade, 'fee');
-        if (feeCost !== undefined) {
+        const feeCostString = this.safeString (trade, 'fee');
+        if (feeCostString !== undefined) {
             const feeCurrencyId = this.safeString (trade, 'feeCurrency');
             const feeCurrencyCode = this.safeCurrencyCode (feeCurrencyId);
             fee = {
-                'cost': feeCost,
+                'cost': feeCostString,
                 'currency': feeCurrencyCode,
-                'rate': this.safeNumber (trade, 'feeRate'),
+                'rate': this.safeString (trade, 'feeRate'),
```

Comment:
@kroitor would you mind having a look at this again?

in my understanding of the [docs](https://docs.ccxt.com/en/latest/manual.html?highlight=fee#fee-structure) - rate is supposed to be a number.

By using `safeString` here, i'm now getting rate as string (for FTX - using python):

output of `fetch_my_trades`:

``` output
  [{'info': {'id': '5848322477',
   'market': 'DOGE/USD',
   'future': None,
   'baseCurrency': 'DOGE',
   'quoteCurrency': 'USD',
   'type': 'order',
   'side': 'buy',
   'price': '0.1872',
   'size': '4952.0',
   'orderId': '108068071023',
   'time': '2021-12-26T13:54:09.398657+00:00',
   'tradeId': '2901586663',
   'feeRate': '0.0',
   'fee': '0.0',
   'feeCurrency': 'DOGE',
   'liquidity': 'maker'},
  'timestamp': 1640526849398,
  'datetime': '2021-12-26T13:54:09.398Z',
  'symbol': 'DOGE/USD',
  'id': '5848322477',
  'order': '108068071023',
  'type': None,
  'takerOrMaker': 'maker',
  'side': 'buy',
  'price': 0.1872,
  'amount': 4952.0,
  'cost': 927.0144,
  'fee': {'cost': 0.0, 'currency': 'DOGE', 'rate': '0.0'},   <-- rate should be a number in my understanding
  'fees': [{'currency': 'DOGE', 'cost': 0.0, 'rate': '0.0'}]}]
```

Reply:
@xmatthias will debug and fix it shortly, thx for reporting!